service/model/actions.go

```diff
@@ -171,12 +172,16 @@ func assertionHeaders(w http.ResponseWriter, user datastore.User, apiCall bool,
 	}
 
 	// Check that the user has permissions to access the model
-	_, err = datastore.Environ.DB.GetAllowedModel(assert.ModelID, user)
+	model, err := datastore.Environ.DB.GetAllowedModel(assert.ModelID, user)
 	if err != nil {
 		log.Println(err)
 		response.FormatStandardResponse(false, "error-get-model", "", err.Error(), w)
 		return
 	}
+	// Increment the revision before it is stored in the database
```

Comment:
I don't understand this change, afaict from the callers assert is decoded from the JSON from the client(s), so this just increment by one what we got from the client instead of considering what we have in the db?